bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/handler/ICloudDeviceHandler.java

```diff
@@ -105,6 +105,8 @@ public void initialize() {
                 if (bridge.getStatus() == ThingStatus.ONLINE) {
                     handler.refreshData();
                     updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
```

Comment:
Can you check if this is still needed with the latest changes in the bridge handler? Perhaps it should be UNKNOWN here as bridge will probably also be UNKNOWN. If bridge is OFFLINE, I believe child handlers will be automatically OFFLINE with BRIDGE_OFFLINE when not having provided a custom implementation of `bridgeStatusChanged`:

https://github.com/openhab/openhab-core/blob/e420cf16476ed6b08b8fab2626ce6d08beecd995/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/BaseThingHandler.java#L643-L651

Reply:
Yes, it is necessary. Otherwise the devices stuck in "INITIALIZING" when the bridge is online during device init. Then they will never init again, even if the bridge goes ONLINE.

Reply:
@maihacke - FYI in case you didn't notice this comment.

Reply:
@jlaur We both commented in the same second ;-)

Reply:
OK, for this last minute bugfix, let's keep it like this then - it can be improved later.